pkg/apis/resource/validation/validation.go

```diff
@@ -819,3 +824,37 @@ func validateNetworkDeviceData(networkDeviceData *resource.NetworkDeviceData, fl
 		fldPath.Child("ips"))...)
 	return allErrs
 }
+
+// AuthorizedForAdmin checks if the request is authorized to get admin access to devices
+// based on namespace label
+func AuthorizedForAdmin(ctx context.Context, deviceRequests []resource.DeviceRequest, namespaceName string, nsClient v1.NamespaceInterface) field.ErrorList {
+	var allErrs field.ErrorList
+	adminRequested := false
+
+	if !utilfeature.DefaultFeatureGate.Enabled(features.DRAAdminAccess) {
+		// No need to validate unless feature gate is enabled
+		return allErrs
+	}
+
+	for i := range deviceRequests {
+		value := deviceRequests[i].AdminAccess
+		if value != nil && *value {
+			adminRequested = true
+			break
+		}
+	}
+	if !adminRequested {
+		// No need to validate unless admin access is requested
+		return allErrs
+	}
```

Comment:
nit: use [slices.IndexFun](https://pkg.go.dev/slices#IndexFunc).

Reply:
thanks for the suggestion. I would like to keep this as is to be consistent with all other functions in validation.go to return the collected errors, instead of finding an index within allErrs.